src/rgw/rgw_datalog.h

```diff
@@ -303,6 +309,8 @@ class RGWDataChangesBE : public boost::intrusive_ref_counter<RGWDataChangesBE> {
   virtual int trim(int index, std::string_view marker,
 		   librados::AioCompletion* c) = 0;
   virtual std::string_view max_marker() const = 0;
+  // 1 on empty, 0 on non-empty, negative o error.
+  virtual int is_empty() = 0;
```

Comment:
consider making this const?

Reply:
It makes a bunch of RADOS calls and stuff, though I could make the IoCtx mutable.

Reply:
ok nevermind!